File: solutions/python/resistor-color/1/resistor_color.py
COLOR_CODES = {
    "black": 0,
    "brown": 1,
    "red": 2,
    "orange": 3,
    "yellow": 4,
    "green": 5,
    "blue": 6,
    "violet": 7,
    "grey": 8,
    "white": 9,
}


def color_code(color: str) -> int:
    return COLOR_CODES[color]


def colors() -> list[str]:
    return list(COLOR_CODES.keys())

File: solutions/c/resistor-color/1/resistor_color.c
#include "resistor_color.h"

resistor_band_t COLORS[] = {
    BLACK,
    BROWN,
    RED,
    ORANGE,
    YELLOW,
    GREEN,
    BLUE,
    VIOLET,
    GREY,
    WHITE,
};

int color_code(resistor_band_t b)
{
    return b;
}

resistor_band_t *colors(void)
{
    return COLORS;
}
